AdblockPlusSafari/DialogPresenterController.m

```diff
@@ -0,0 +1,266 @@
+/*
+ * This file is part of Adblock Plus <https://adblockplus.org/>,
+ * Copyright (C) 2006-2015 Eyeo GmbH
+ *
+ * Adblock Plus is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * Adblock Plus is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Adblock Plus.  If not, see <http://www.gnu.org/licenses/&gt.
+ */
+
+
+#import "DialogPresenterController.h"
+
+#import "RootController.h"
+
+typedef NS_ENUM(NSUInteger, DialogControllerType) {
+  DialogControllerTypeFirst,
```

Comment:
The ordinal numbering is a useful information, but not enough alone. Makes sense only if the related change request is known and studied. The controller names should contain a related context.
First = 1AcceptableAdsExplanation
Second = 2ActivationHowtoVideo
Third = 3ActivationWaiting
Fourth = 4ActivationConfirmation

Obviously applies to the class names and related function names too.